core/sql/optimizer/SynthType.cpp

```diff
@@ -1010,7 +1010,81 @@ const NAType *BuiltinFunction::synthesizeType()
 	    SQLChar(maxLength, typ1.supportsSQLnull());
       }
     break;
+ 
+    case ITM_ISIPV4:
+    case ITM_ISIPV6:
+      {
+        // type cast any params
+        ValueId vid1 = child(0)->getValueId();
+        SQLChar c1(ComSqlId::MAX_QUERY_ID_LEN);
+        vid1.coerceType(c1, NA_CHARACTER_TYPE);
+        //input type must be string
+        const NAType &typ1 = child(0)->getValueId().getType();
+
+        if (typ1.getTypeQualifier() != NA_CHARACTER_TYPE)
+          {
+	    *CmpCommon::diags() << DgSqlCode(-4045) << DgString0("IS_IP");
+	    return NULL;
+          }
+        retType = new HEAP
+           SQLSmall(TRUE, FALSE);
```

Comment:
SQLSmall is a 2-byte integer. Wouldn't you want a 4-byte integer at least for ipv4 and an 8-byte integer for ipv6?